graphite-project/graphite-web
# [1.1.x] handle exceptions if params cannot be type converted (#2547) | fix minor bug in query param evaluation (#2547) | fails to instantiate find query must be due to bad user input (#2547) | add tests for invalid parameter types (#2547) |

Backports the following commits to 1.1.x:
 - handle exceptions if params cannot be type converted (#2547)
 - fix minor bug in query param evaluation (#2547)
 - fails to instantiate find query must be due to bad user input (#2547)
 - add tests for invalid parameter types (#2547)
 - fix formatting (#2547)
 - test validator with default values (#2555)
 - use default value in validator when necessary (#2555)
 - flake8: include contrib/ subdir (#2554)
 - flake8: re-enable rule E713 - use "not in" (#2554)
 - make consolidation func avg alias for average (#2556)
 - move all validation into Param.validateValue (#2557)
 - move check acceptance of optional params up (#2557)
 - flake8: re-enable E122,E124 (indent of continuation and closing bracket) (#2558)
 - flake8: re-enable F841 (local variable assigned but not used) (#2559)
